### PR TITLE
Fix divide by zero when shutting down Typha if it has no clients.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Exit cleanly instead of panicking.
## Related issues/PRs

Fixes the panic seen in #7553 (but the overall issue may be something else)

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Fix a divide-by-zero panic in Typha if it received a SIGTERM when it had no active connections.  Since Typha exits in either case, the impact was limited to a scary panic log.
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Review by sridhartigera — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by mgleung** [comment]
@fasaxc since we cut release branches, I think we'll need a cherry-pick of this to the `release-v3.26` branch

**Pull request merged**
